lib/config-manager.coffee

```diff
@@ -9,18 +9,31 @@ module.exports = ConfigManager =
     fileStoragePath: path.join(__dirname, '..', 'kernel-configs')
 
     writeConfigFile: (onCompleted) ->
-        try
-            fs.mkdirSync(@fileStoragePath)
-        catch e
-            if e.code != 'EEXIST'
-                throw e
-        filename = 'kernel-' + uuid.v4() + '.json'
-        portfinder.findMany 5, (ports) =>
-            config = @buildConfiguration ports
-            configString = JSON.stringify config
-            filepath = path.join(@fileStoragePath, filename)
-            fs.writeFileSync filepath, configString
-            onCompleted filepath, config
+        customKernelConnectionPath = path.join atom.project.rootDirectories[0].path, 'hydrogen', 'kernel', 'hydrogen-kernel.json'
+        fs.access customKernelConnectionPath, (err) =>
+            if err?
+                try
+                    console.log "fileStoragePath: ", @fileStoragePath
+                    fs.mkdirSync(@fileStoragePath)
+                catch e
+                    if e.code != 'EEXIST'
+                        throw e
+                filename = 'kernel-' + uuid.v4() + '.json'
+                portfinder.findMany 5, (ports) =>
+                    config = @buildConfiguration ports
+                    configString = JSON.stringify config
+                    filepath = path.join(@fileStoragePath, filename)
+                    fs.writeFileSync filepath, configString
+                    console.log "Creating kernel connection: ", filepath
+                    onCompleted filepath, config
+            else
+                fs.readFile customKernelConnectionPath, 'utf8', (err, data) ->
+                    unless err?
+                        config = JSON.parse data
+                        console.log "Using custom kernel connection: ", customKernelConnectionPath
+                        atom.notifications.addInfo 'Custom kernel connection:',
+                            detail: "Make sure to manually start the custom kernel that serves the connection in #{customKernelConnectionPath}", dismissable: true
```

Comment:
I removed it in the latest version, nevertheless, I forgot to mention why I added it.

If a user starts with a custom kernel, with the custom connection, and then triggers "Hydrogen: Restart Kernel", the shutdown signal will be sent to the external (Docker) kernel, which will stop the kernel and remove the file. And then, Hydrogen will "restart" the kernel, but as there's not a custom kernel connection anymore it will start a new, local, fresh kernel.

Using my example from the README, this will make the package `markdown` import while using the kernel container, but throw an error after restarting the kernel (using the local kernel).

I just wanted to make sure to put it here, in case you want to get the notification (auto-dismissed) back or do something else for that "corner case".